device/tt_device.h

```diff
@@ -585,7 +585,7 @@ class tt_device
     std::unordered_map<chip_id_t, tt_SocDescriptor> soc_descriptor_per_chip = {};
 };
 
-#include "device/architecture_implementation.h"
+#include "device/tt_device/tt_device.h"
```

Comment:
`device/tt_device.h` and `device/tt_device/tt_device.h` is confusing.

Reply:
Yes, exactly. I've left a comment in the PR because of that, to wait for tt_device rename to cluster.h first maybe. Even tt_device and TTDevice is confusing. @vtangTT removed this class previously specifically to remove confusion.
I'll leave this comment open until this is resolved.